### ISSUE TITLE
`collapseWhitespace` changes attributes

### ISSUE THREAD

**Issue by remcohaszing** [kangax/html-minifier#117]
When trying to minify the following template taken from [angular-ui-bootstrap](https://github.com/angular-ui/bootstrap/blob/master/template/alert/alert.html) using collapseWhitespaces

``` html
<div class='alert' ng-class='"alert-" + (type || "warning")'>
    <button ng-show='closeable' type='button' class='close' ng-click='close()'>&times;</button>
    <div ng-transclude></div>
</div>
```

The following is the result:

``` html
<div class="alert" ng-class="\"alert-\" + (type || \"warning\")"><button ng-show="closeable" type="button" class="close" ng-click="close()">&times;</button><div ng-transclude=""></div></div>
```

This contains two bugs:
- The ng-class attribute is no longer valid;
- The boolean attribute ng-transclude has a new value, empty string.

**Comment by jeremypeter** [comment]
You can fix the first bug by changing your single quotes to double, and double quotes to singles

``` html
<div class="alert" ng-class="'alert-' + (type || 'warning')">
    <button ng-show="closeable" type="button" class="close" ng-click="close()">&times;</button>
    <div ng-transclude></div>
</div>
```

**Comment by kangax** [comment]
@remcohaszing Why isn't ng-class valid? From what I can see, html-minifier is doing everything properly — it just normalizes quotes and escapes inner ones.

As for "ng-transclude", html-minifier doesn't know anything about it; doesn't know that it's supposed to be treated as boolean attributes (meaning that it's safe to remove its value).

**Comment by remcohaszing** [comment]
This is not my own code. I noticed the angular-ui team did not minify their angular templates, so I decided to try it myself. It did make the tests fail.

@jeremypeter Your fix does work, but it shouldn't make a difference.
@kangax I'm new at this, so correct me if I'm wrong, but using a backslash is not the correct way to escape quotes in html.
When trying to minify the following template, even html-minifier throws a Parse Error:

``` html
<div class="\" "></div>
```

The error is triggered by using an escaped quote followed by a space.
